solutions/cses-1625.mdx

```diff
@@ -0,0 +1,281 @@
+---
+id: cses-1625
+source: CSES
+title: Grid Paths
+author: Vivian Han
+---
+
+## Explanation
+
+Adapted from https://usaco.guide/CPH.pdf#page=61
+
+<Warning title="Run times">
+Please note: the running times and number of recursive calls listed are for a separate but very similar problem (# paths from upper left to *lower right* corner on a 7x7 grid). 
+They may not be entirely accurate, but they illustrate the effect of each optimization. 
+The data is also C++ specific, so Java and Python users will experience significantly different runtimes.
+The consequences of this discrepancy which will be addressed later in the explanation.
+</Warning>
+
+### Basic Algorithm
+
+The first version of the algorithm does not contain any optimizations. We simply
+use backtracking to generate all possible paths from the upper-left corner to
+the lower-right corner and count the number of such paths.
+
+- running time: 483 seconds
+-  number of recursive calls: 76 billion
+
+### Optimization 1
+
+If the path reaches the lower-right square before it has visited all other
+squares of the grid, it is clear that it will not be possible to complete the
+solution. Using this observation, we can terminate the search immediately if we
+reach the lower-right square too early.
+
+- running time: 119 seconds
+- number of recursive calls: 20 billion
+
+### Optimization 2
+
+If the path touches a wall and can turn either left or right, the grid splits
+into two parts that contain unvisited squares. In this case, we cannot visit all
+squares anymore, so we can terminate the search.
+
+- running time: 1.8 seconds
+- number of recursive calls: 221 million
+
+### Optimization 3
+
+The idea of Optimization 3 can be generalized: if the path cannot continue
+forward but can turn either left or right, the grid splits into two parts that
+both contain unvisited squares. It is clear that we cannot visit all squares
+anymore, so we can terminate the search.
+
+- running time: 0.6 seconds
```

Comment:
would it make sense to capitalize the first word for all lists?

also it would be nice but not required to texify the #

Reply:
e.g 0.6 in this case